pdns/logger.cc

```diff
@@ -47,14 +47,17 @@ void Logger::log(const string &msg, Urgency u)
 #ifndef RECURSOR
   bool mustAccount(false);
 #endif
-  struct tm tm;
-  time_t t;
-  time(&t);
-  tm=*localtime(&t);
-
   if(u<=consoleUrgency) {
     char buffer[50];
-    strftime(buffer,sizeof(buffer),"%b %d %H:%M:%S ", &tm);
+    buffer[0] = '\0';
+    if(!d_disableTimeStamps) {
+      struct tm tm;
+      time_t t;
+      time(&t);
+      tm=*localtime(&t);
+
+      strftime(buffer,sizeof(buffer),"%b %d %H:%M:%S ", &tm);
```

Comment:
Unrelated to this patch but we should, theoretically, be checking the return value of `strftime`.